### ISSUE TITLE
Polar H10 not detected by OpenTracks on a Huawei P40 lite

### ISSUE THREAD

**Issue by hg-runs** [OpenTracksApp/OpenTracks#593]
**Describe the bug**
Tried to use a Polar H10 heart rate monitor (which is listed as usable by OpenTracks).
It's detected by the smartphone but not within the app.

**Am I missing something here?**

**Screenshots**
![Wire 2021-02-08 at 14-36-23](https://user-images.githubusercontent.com/29901271/107233979-75577e80-6a23-11eb-9188-5accdaabb6f3.jpg)
![Wire 2021-02-08 at 14-36-16](https://user-images.githubusercontent.com/29901271/107233986-77b9d880-6a23-11eb-89ce-1e7a55a0605f.jpg)
![Wire 2021-02-08 at 14-36-27](https://user-images.githubusercontent.com/29901271/107233964-71c3f780-6a23-11eb-87b4-d69c21eb737b.jpg)

**Technical information**
 - Device: Huawei P40 lite
 - OS: Android version 10

**Comment by dennisguse** [comment]
Nice detailed report!

TLDR: just try it again with the H10 on your body.
The problem is that the H10 is turned off (aka powersaving mode).
OpenTracks does not check which sensors are paired with your phone, but searches for sensors how are actively announcing that they are there.
And the H10 only turns on, when it is attached to the body (i.e., it's sensor have contact with something that is conductive).

Other thing that might happen: the battery of the H10 is empty.

PS: I used an H10 to implement the HR functionality :)

**Comment by hg-runs** [comment]
Thanks for the quick answer.

Back with some more information:

- The H10 was new. I've changed the battery though but no luck
- The H10 was not paired
- The H10 was attached to my body - other wise it would not have been listed in the bluetooth discovery of the phone

Why the smartphone's bluetooth discovery is working but not the app's one?

Obviously I'm interested to get this running.
Anything else I can test/do?

**Comment by dennisguse** [comment]
Can you check that either FitoTrack or RunnerUp can access the HR sensor?

Some more details:
In the settings, OpenTracks searches for (HR) sensors.
This means that the sensors must be turned on, not connected to any device (otherwise they don't announce that they are available), and in their announcement package announce that the HR gatt is provided.
A Miband doesnt do the later and we have a workaround in place for this.

And we tell Android to search for HR sensors (filter by serviceId): https://github.com/OpenTracksApp/OpenTracks/blob/e8017a4e9d8c4c420d9c1663dc40a4a036d77f49/src/main/java/de/dennisguse/opentracks/settings/bluetooth/BluetoothLeSensorPreference.java#L184

**Comment by hg-runs** [comment]
Just FYI:
Using this version of OpenTracks the Polar H10 all of sudden is recognized on a Huawei P40 lite.
But not only the HRM but also a GARMIN speed sensor 2 and a cadence sensor 2 (which weren't working with any other previous versions of OpenTracks).

Neither sure what has been changed nor had a look at the repository code. Fitotrack and RunnerUp are still not recognizing the Polar H10
![Wire 2021-05-05 at 08-07-08](https://user-images.githubusercontent.com/29901271/117113042-3c147d80-ad8a-11eb-8b97-5eff3d91b384.jpg)

![Wire 2021-05-05 at 08-08-02](https://user-images.githubusercontent.com/29901271/117113063-420a5e80-ad8a-11eb-9fd7-40b0abfe3380.jpg)

**Comment by dennisguse** [comment]
@hg-runs Great!
i would really love to know why as I don't recall any change.
Whatever - glad that it is working and really cool that you use all these sensors!
